numpyro/distributions/directional.py

```diff
@@ -0,0 +1,120 @@
+import jax.numpy as np
```

Comment:
I'll add the license :) 

Circular is an apt name because it refers to the circular boundary, and it has the added benefit of consistency. However, the von Mises, [wrapped Cauchy](https://en.wikipedia.org/wiki/Wrapped_Cauchy_distribution), and Johnson-Wehrly distributions are distributions over directions (and not circles), just like discrete are distributions over discrete variables and continuous are over continuous variables. Which is why I think directional is a more appropriate name.

Reply:
Yeah, thanks for explaining! It is great to know about the reason behind the chosen name. That makes a lot of sense to me.